src/Symfony/Component/Serializer/Normalizer/CollectionDenormalizer.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Normalizer;
+
+use Symfony\Component\PropertyInfo\Type;
+use Symfony\Component\Serializer\Exception\BadMethodCallException;
+use Symfony\Component\Serializer\Exception\NotNormalizableValueException;
+
+/**
+ * Denormalizes arrays of objects into a single collection.
+ *
+ * @author Nikita Shipilov <nikita.shipilov.22@gmail.com>
+ *
+ * @final
+ */
+class CollectionDenormalizer implements DenormalizerInterface, DenormalizerAwareInterface, CacheableSupportsMethodInterface
```

Comment:
I'd have enhanced `ArrayDenormalizer` instead of creating a new denormalizer, WDYT?